### DISCUSSION TITLE
Builtin highpass and lowpass audio filters

### DISCUSSION THREAD

**Discussion by tphakala** [tphakala/birdnet-go#286] [Announcements]
Latest container images have built-in highpass and lowpass audio filter features. The highpass filter is useful for attenuating AC hum or any other low-frequency noise which may be interfering with your audio capture. Please be careful not to set the highpass filter to too high a frequency or the lowpass filter to too low a frequency, as these filters apply to both the BirdNET analyzer-fed audio and the audio capture.

Filters are available in Settings / Audio Capture section

https://github.com/user-attachments/assets/8ffc2188-0d19-45c4-82ed-2c2ade4d6ecb

**Comment by PovilasID** [comment]
Very cool but can you or users share settings that work for them?
Also how to debug if it is not too low or high?

**Reply by tphakala** [reply]
It depends on what you want to detect and the type of interference in your environment. Some owl calls are very low frequency, while songbirds are on the higher end of the frequency scale.

As a rule of thumb, I would avoid setting the high-pass filter above 400 Hz and the low-pass filter below 10,000 Hz. AC hum frequency is typically 50 or 60 Hz, so if you're experiencing issues with that, a proper high-pass filter cut-off frequency would be around 60 Hz with sufficient attenuation to silence the hum.

**Reply by alexbelgium** [reply]
Hi, what you can do also is download previously recorded birdsongs, then pass it through the Birdnet-Analyzer app. The app allows to dynamically set high/low pass filters while performing the analysis, which will then show you the direct impact of the values on the species that you intend to target. You can't mention the filter parameters in the Birdnet-Analyzer app, but it should already provide you a good idea.

Location of the setting :
<img width="300" alt="image" src="https://github.com/user-attachments/assets/0f8f868e-7717-4c0f-83c4-0a06ab57ff48">

Don't forget that the latest birdnet model is trained from 0 to 15khz, so already anything above 15khz is not useful.

**Comment by juicees** [comment]
Hey. Cant add(save) audio filters anymore in latest ":dev:" version. It says "Please fill out all required fields." even i have been filling all fields many times. Before, some weeks ago it was still working like a charm. Any suggestions?

**Reply by tphakala** [reply]
this is fixed in latest dev image
